src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,75 @@
+package mate.academy.dao.impl;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.MovieSessionDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.MovieSession;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(movieSession);
+            transaction.commit();
+            return movieSession;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't to add a movie session " + movieSession, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Optional<MovieSession> get(Long id) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> movieSessionQuery = session.createQuery(
+                    "FROM MovieSession ms "
+                            + "LEFT JOIN fetch ms.movie "
+                            + "LEFT JOIN fetch ms.cinemaHall "
+                            + "WHERE ms.id = :id", MovieSession.class);
+            movieSessionQuery.setParameter("id", id);
+            return movieSessionQuery.uniqueResultOptional();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get a movie session by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            LocalDateTime startOfDay = date.atStartOfDay();
+            LocalDateTime endOfDay = date.atTime(23, 59, 59);
+            Query<MovieSession> movieSessionQuery = session.createQuery(
+                    "FROM MovieSession ms "
+                            + "LEFT JOIN fetch ms.movie "
+                            + "LEFT JOIN fetch ms.cinemaHall "
+                            + "WHERE ms.movie.id = :movieId "
+                            + "AND ms.showTime BETWEEN :startOfDay AND :endOfDay",
+                    MovieSession.class);
+            movieSessionQuery.setParameter("movieId", movieId);
+            movieSessionQuery.setParameter("startOfDay", startOfDay);
+            movieSessionQuery.setParameter("endOfDay", endOfDay);
```

Comment:
Let's better to use the next approach for checking the date values:
```
 + "and year(ms.showtime) = :year "
                            + "and month(ms.showtime) = :month "
                            + "and day(ms.showtime) = :day",
                    MovieSession.class);
            query.setParameter("year", date.getYear());
            query.setParameter("month", date.getMonthValue());
            query.setParameter("day", date.getDayOfMonth());
```
So you don't need to introduce  variables `startOfDay` and `endOfDay`